zipline/data/minute_bars.py

```diff
@@ -1242,3 +1246,94 @@ def load_raw_arrays(self, fields, start_dt, end_dt, sids):
 
             results.append(out)
         return results
+
+
+class MinuteBarUpdateReader(with_metaclass(ABCMeta, object)):
+    """
+    Abstract base class for minute update readers.
+    """
+
+    @abstractmethod
+    def read(self, dts, sids):
+        """
+        Read and return pricing update data.
+
+        Parameters
+        ----------
+        dts : DatetimeIndex
+            The minutes for which to read the pricing updates.
+        sids : iter[int]
+            The sids for which to read the pricing updates.
+
+        Returns
+        -------
+        data : iter[(int, DataFrame)]
+            Returns an iterable of ``sid`` to the corresponding OHLCV data.
+        """
+        raise NotImplementedError
+
+
+class H5MinuteBarUpdateWriter(object):
+    """
+    Writer for files containing minute bar updates for consumption by a writer
+    for a ``MinuteBarReader`` format.
+
+    Parameters
+    ----------
+    path : str
+        The destination path.
+    complevel : int
+        The HDF5 complevel, defaults to ``5``.
+    complib : str
+        The HDF5 complib, defaults to ``zlib``.
+    """
+
+    FORMAT_VERSION = 0
+
+    _COMPLEVEL = 5
+    _COMPLIB = 'zlib'
+
+    def __init__(self, path, complevel=None, complib=None):
+        self._complevel = complevel if complevel \
+            is not None else self._COMPLEVEL
+        self._complib = complib if complib \
+            is not None else self._COMPLIB
+        self._path = path
+
+    def write(self, frames):
+        """
+        Write the frames to the target HDF5 file, using the format used by
+        ``pd.Panel.to_hdf``
+
+        Parameters
+        ----------
+        frames : iter[(int, DataFrame)] or dict[int -> DataFrame]
+            An iterable or other mapping of sid to the corresponding OHLCV
+            pricing data.
+        """
+        _store = HDFStore(
+            self._path, 'w', complevel=self._complevel, complib=self._complib)
+        panel = pd.Panel.from_dict(dict(frames))
+        panel.to_hdf(_store, 'updates')
+        _store.close()
+        h5file = tables.open_file(self._path, mode='r+')
+        metadata = h5file.create_group('/', 'metadata')
```

Comment:
Could not find a way via HDFStore to set it on the file.
However looked through the docs for `h5file.File` again and found `set_node_attr` so using this instead:
```
with tables.open_file(self._path, mode='r+') as h5file:
       h5file.set_node_attr('/', 'version', 0)
```